tests/phpunit/unit/EventListener/StorageEventListenerTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Bolt\Tests\EventListener;
+
+use Bolt\Storage\Entity\Users;
+use Bolt\Storage\EntityManager;
+use Bolt\Storage\Database\Schema\Manager;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Bolt\Logger\FlashLoggerInterface;
+use PasswordLib\Password\Factory as PasswordFactory;
+use Bolt\EventListener\StorageEventListener;
+use Bolt\Events\StorageEvent;
+use Bolt\Config;
+use Prophecy\Argument;
+use PasswordLib\Password\Implementation\Blowfish;
+
+class StorageEventListenerTest extends \PHPUnit_Framework_TestCase
+{
+    private $user;
+    private $entityManager;
+    private $config;
+    private $manager;
+    private $urlGenerator;
+    private $flashLogger;
+    private $passwordFactory;
+    private $listener;
+    private $storageEvent;
+
+    public function setUp()
+    {
+        $this->user = $this->prophesize(Users::class);
+        $this->entityManager = $this->prophesize(EntityManager::class);
+        $this->config = $this->prophesize(Config::class);
+        $this->manager = $this->prophesize(Manager::class);
+        $this->urlGenerator = $this->prophesize(UrlGeneratorInterface::class);
+        $this->flashLogger = $this->prophesize(FlashLoggerInterface::class);
+        $this->passwordFactory = $this->prophesize(PasswordFactory::class);
+
+        $this->listener = new StorageEventListener(
+            $this->entityManager->reveal(),
+            $this->config->reveal(),
+            $this->manager->reveal(),
+            $this->urlGenerator->reveal(),
+            $this->flashLogger->reveal(),
+            $this->passwordFactory->reveal(),
+            5
```

Comment:
ok cool. I wil try adapt this to the standard form from the other tests and use this code in an RFC